src/Peers/provider.cpp

```diff
@@ -199,64 +205,35 @@ void Provider::processWorkload() {
     proto.set_training_data_index_filename(indexFile);
     string serialized;
     proto.SerializeToString(&serialized);
-    zmq_sender.send(serialized);
+    ml_zmq_sender.send(serialized);
     cout << "Waiting for processed data..." << endl;
-    auto rcvdData = zmq_receiver.receive();
-
-    // turn the string back into a vector
-    vector<int> data;
-    string num = "";
-    for (int i = 0; i < rcvdData.size(); i++) {
-        if (rcvdData[i] == ',') {
-            data.push_back(stoi(num));
-            num = "";
-        } else {
-            num += rcvdData[i];
-        }
-    }
-    data.push_back(stoi(num));
-    cout << "Processed workload with data: ";
-    for (int i = 0; i < data.size(); i++) {
-        cout << data[i] << " ";
-    }
-    cout << endl;
+    auto rcvdData = ml_zmq_receiver.receive();
+    cout << "Received processed data" << endl;
 
     // store result in taskResponse->trainingdata
-    taskResponse = make_unique<TaskResponse>(data);
+    taskResponse = make_unique<TaskResponse>(rcvdData);
     cout << "Completed assigned workload" << endl;
 }
 
-TaskResponse Provider::aggregateResults(vector<vector<int>> followerData) {
-    vector<int> curr_data = taskResponse->getTrainingData();
+TaskResponse Provider::aggregateResults(vector<string> followerData) {
+    // append leader's data to followerData
+    string curr_data = taskResponse->getTrainingData();
     followerData.push_back(curr_data);
 
-    vector<int> indicies(followerData.size(), 0);
-    vector<int> data;
-
-    while (true) {
-        int minVal = INT_MAX;
-        int minIdx = -1;
-        for (int i = 0; i < followerData.size(); i++) {
-            if (indicies[i] < followerData[i].size() &&
-                followerData[i][indicies[i]] < minVal) {
-                minVal = followerData[i][indicies[i]];
-                minIdx = i;
-            }
-        }
-
-        if (minIdx == -1) {
-            break;
-        }
-
-        data.push_back(minVal);
-        indicies[minIdx]++;
+    // send each follower's data to the aggregator
+    for (int i = 0; i < followerData.size(); i++) {
+        payload::AggregatorInputData proto;
```

Comment:
wouldn't the cpp side be receiving the same proto from python code? So can't we set the proto equal to the python serialized proto? Or just keep it as a string.

Reply:
Like we recieved a serialized proto and then we are serializing it again where the serialized proto is now in the bytes field?